lightning/src/routing/router.rs

```diff
@@ -829,6 +831,20 @@ where L::Target: Logger {
 	// - when we want to stop looking for new paths.
 	let mut already_collected_value_msat = 0;
 
+	// We only consider paths shorter than our maximum length estimate.
+	// In the legacy onion format, the maximum number of hops used to be a fixed value of 20.
+	// However, in the TLV onion format, there is no fixed maximum length, but the `hop_payloads`
+	// field is always 1300 bytes. As the `tlv_payload` for each hop may vary in length, we have to
+	// estimate how many hops the route may have so that it actually fits the `hop_payloads` field.
+	//
+	// We estimate 3+32 (payload length and HMAC) + 2+8 (amt_to_forward) + 2+4 (outgoing_cltv_value) +
+	// 2+8 (short_channel_id) = 61 bytes for each intermediate hop and 3+32
+	// (payload length and HMAC) + 2+8 (amt_to_forward) + 2+4 (outgoing_cltv_value) + 2+32+8
+	// (payment_secret and total_msat) = 93 bytes for the final hop.
+	// Since the length of the potentially included `payment_metadata` is unkown to us, we round
+	// down from (1300-93) / 61 = 19.78... to arrive at a conservative estimate of 19.
+	const MAX_PATH_LENGTH_ESTIMATE: u8 = 19;
```

Comment:
I'm a bit confused whether this is the maximum path length *including* the final hop or *excluding* the final hop? Seems to be excluding the final hop when arriving at `19`, but when it's used it seems to limit the  path length including the final hop

Don't think it matters in the end, but just confused me a bit at first

Reply:
AFAIK it should limit the *entire* path, aka it can include a number of hops less or equal to `MAX_PATH_LENGTH_ESTIMATE`. Clarified that wording in https://github.com/lightningdevkit/rust-lightning/pull/1476/commits/1408aa79db8f29a8ebff7041e206fce4d476c200.